src/UseCase/FetchIssueUseCase.php

```diff
@@ -32,11 +36,17 @@ public function execute(string $issueName, array $issueNameFilter = []): ?Issue
             /** @var Issue $issue */
             $issue = $this->issueService->get($issueName, ['fields' => ['status']]);
         } catch (JiraException $e) {
+            if ($e->getCode() === 404) {
+                throw new JiraTicketNotFoundException(
+                    'Jira ticket does not exist or was deleted',
+                    $e->getCode()
+                );
```

Comment:
```suggestion
                throw new JiraTicketNotFoundException(
                    'Jira ticket does not exist or was deleted',
                    $e->getCode(),
                    $e
                );
```

As a small rule of thumb. If you catch a system foreign exception and want to rethrow it with you system internal one. Always add the previous exception to the stacktrace or you will hide the source exception which makes it harder to track down problems.